test/service/s3_service_test.rb

```diff
@@ -30,6 +30,20 @@ class ActiveStorage::Service::S3ServiceTest < ActiveSupport::TestCase
       assert_match /#{SERVICE_CONFIGURATIONS[:s3][:bucket]}\.s3.(\S+)?amazonaws.com.*response-content-disposition=inline.*avatar\.png/,
         @service.url(FIXTURE_KEY, expires_in: 5.minutes, disposition: :inline, filename: "avatar.png")
     end
+
+    test "encrypt file with server_side_encryption upload option" do
+      skip "server_side_encryption option not supplied " unless @service.upload_options[:server_side_encryption]
```

Comment:
Let’s construct a service with the `:server_side_encryption` option populated in this test.